gnovm/pkg/gnolang/preprocess.go

```diff
@@ -822,100 +831,140 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 								Op:    n.Op,
 								Right: rn,
 							}
-							resn := Node(Call(tx, n2))
+							resn := Node(Call(tx, n2)) // this make current node to gonative{xxx}
 							resn = Preprocess(store, last, resn)
 							return resn, TRANS_CONTINUE
 							// NOTE: binary operations are always computed in
 							// gno, never with reflect.
 						} else {
-							if isShift {
-								// nothing to do, right type is (already) uint type.
-								// we don't yet know what this type should be,
-								// but another checkOrConvertType() later does.
-								// (e.g. from AssignStmt or other).
-							} else {
-								// convert n.Left to right type.
-								checkOrConvertType(store, last, &n.Left, rt, false)
-							}
+							// convert n.Left to right type.
+							checkOrConvertType(store, last, &n.Left, rt, false)
 						}
-					} else if lcx.T == nil {
+					} else if lcx.T == nil { // LHS is nil.
 						// convert n.Left to typed-nil type.
 						checkOrConvertType(store, last, &n.Left, rt, false)
+					} else { // left is typed const, right not const
+						if isUntyped(rt) { // e.g. int(1) + 1<<x
+							checkOrConvertType(store, last, &n.Right, lt, false)
+						} else { // both typed, left typed const, right typed non-const
+							checkOrConvertType(store, last, &n.Left, rt, false) // see eql_0a1a0.gno
```

Comment:
should rather be, 'lt.TypeID() != rt.TypeID()' and panic. 
when both are typed, there should not be any conversion.